llvm/lib/Transforms/Yk/ModuleClone.cpp

```diff
@@ -9,7 +9,11 @@
 //     versions of such functions in the module: the original and the cloned
 //     version.
 //   - Functions **with** their address taken remain only in their original form
-//     and are not cloned.
+//     and are not cloned. These functions are not cloned because then we would
+//     need to update their references in the callers to point to the newly
+//     cloned function. Since function pointers can be stored in variables,
+//     passed as arguments, etc. tracking and updating all these references is
+//     feasible but complex.
 //
 // - **Cloned Function Naming:**
 //   - The cloned functions are renamed by adding the prefix `__yk_unopt_` to
```

Comment:
Above we are saying that the cloned functions are the optimised ones but here we are contradicting this saying the clones are unoptimised. Remind me again which is which as I keep forgetting.

Reply:
Oops, that a leftover from previous renaming :no_mouth:  
Updated comment :point_right: c10ec59d82f7ca06b8b64d516836a88ae84aa3ba

But generally:

```
// opt
func main(%arg0: i32, %arg1: ptr) -> i32 {
  bb0:
    ...
    call __yk_trace_basicblock_dummy(0i32, 0i32)

// unopt
func __yk_unopt_main(%arg0: i32, %arg1: ptr) -> i32 {
  bb0:
    ...
    call __yk_trace_basicblock(9i32, 0i32)
```